vite-project/src/components/UI/modal/MyModal.tsx

```diff
@@ -1,23 +1,26 @@
 import ClearSearch from '../../../assets/ClearSearch';
+import { setVisible } from '../../../store/cardSlice';
+import { useAppDispatch, useAppSelector } from '../../../store/hook';
 import classes from './MyModal.module.scss';
 
 interface IProps {
-  visible: boolean;
-  setVisible: React.Dispatch<React.SetStateAction<boolean>>;
   children: JSX.Element;
 }
 
-export const MyModal = ({ children, visible, setVisible }: IProps) => {
-  const cl = [classes.myModal];
-  if (visible) {
-    cl.push(classes.active);
-  }
+export const MyModal = ({ children }: IProps) => {
+  const isVisible = useAppSelector((state) => state.cards.isVisible);
+  const dispatch = useAppDispatch();
```

Comment:
better not to use redux hooks in components
Create custom hooks in the store and just take them from there

Reply:
In one hook, the action you need is called